src/bin/rugs_metadata_server.rs

```diff
@@ -216,6 +225,44 @@ mod tests {
         Ok(pool)
     }
 
+    /// Test reading auth config secrets from files
+    #[tokio::test]
+    async fn config_secrets() -> Result<()> {
+        std::fs::write("./rugs_user_auth", USER_AUTH).expect("Could not write user secret to file");
+        std::fs::write("./rugs_ci_auth", CI_AUTH).expect("Could not write CI secret to file");
+
+        let cur_user_auth_file = std::env::var(USER_AUTH_FILE_KEY);
+        let cur_ci_auth_file = std::env::var(CI_AUTH_FILE_KEY);
+
+        std::env::set_var(USER_AUTH_FILE_KEY, "./rugs_user_auth");
+        std::env::set_var(CI_AUTH_FILE_KEY, "./rugs_ci_auth");
+
+        let config = Config::from_env();
+
+        std::fs::remove_file("./rugs_user_auth").expect("Could not delete user secret file");
+        std::fs::remove_file("./rugs_ci_auth").expect("Could not delete CI secret file");
+
+        if let Ok(prev_user_auth_file) = cur_user_auth_file {
+            std::env::set_var(USER_AUTH_FILE_KEY, prev_user_auth_file);
+        } else {
+            std::env::remove_var(USER_AUTH_FILE_KEY);
+        }
+
+        if let Ok(prev_ci_auth_file) = cur_ci_auth_file {
+            std::env::set_var(CI_AUTH_FILE_KEY, prev_ci_auth_file);
+        } else {
+            std::env::remove_var(CI_AUTH_FILE_KEY);
+        }
+
+        assert_eq!(config.user_auth, USER_AUTH);
+        assert_eq!(config.ci_auth, CI_AUTH);
+
+        assert_ne!(config.user_auth, "");
```

Comment:
With the above assert, this is just asserting that `USER_AUTH` and `CI_AUTH` are not empty, and they're constants, so that seems like something we can omit.